pkg/sql/pg_catalog.go

```diff
@@ -680,7 +680,11 @@ CREATE TABLE pg_catalog.pg_constraint (
 					confupdtype = fkActionNone
 					confdeltype = fkActionNone
 					confmatchtype = fkMatchTypeSimple
-					if conkey, err = colIDArrayToDatum(con.Index.ColumnIDs); err != nil {
+					columnIDs := con.Index.ColumnIDs
+					if con.FK.SharedPrefixLen < int32(len(columnIDs)) {
```

Comment:
I thought this check was usually just `!= 0`? In very old table descs, it might be unset, and thus zero, so `< len`  could get us in trouble.

Reply:
Can you point to an example?  Having to check 0 as well is a little strange.

Reply:
Ok, I think this is ready to go.  PTAL

I added a comment on `structured.proto` to reflect the need to check for >0 to use the `sharedprefixlen`.